src/cli/rustup_mode.rs

```diff
@@ -725,6 +726,9 @@ pub(crate) fn cli() -> Command {
                         .value_parser(partial_toolchain_desc_parser),
                 )
                 .arg(Arg::new("topic").help(TOPIC_ARG_HELP))
+                .subcommand(
+                    Command::new("servedoc")
```

Comment:
I think this should probably be `doc --serve` rather than a new command.

Reply:
It should be a new `--serve` flag on the existing `doc` command, that works orthogonally to other flags. Maybe it should take an optional port number or `SocketAddr` (like `192.168.0.1:3112`) as argument?